### PR TITLE
FFWEB-2356: Add arrow_functions

### PR DESCRIPTION
**Pull request by paulfcdd**
- Solves issue: 
https://jira.omikron.net/browse/FFWEB-2356

- Description: 
Convert of lambda functions to be an arrow function

- Tested with Magento editions/versions: 
2.4.2

- Tested with PHP versions: 
7.4.20

### CODE REVIEW

File: src/Controller/Adminhtml/TestConnection/TestFtpConnection.php [hunk 1]
```diff
@@ -55,13 +55,9 @@ private function getConfig(array $params): array
...
 
-        $filtered = array_filter($params, function (string $key) use ($prefix): bool {
-            return (bool) preg_match("#^{$prefix}#", $key);
-        }, ARRAY_FILTER_USE_KEY);
+        $filtered = array_filter($params, fn(string $key) => (bool) preg_match("#^{$prefix}#", $key), ARRAY_FILTER_USE_KEY);
 
-        return array_combine(array_map(function (string $key) use ($prefix): string {
-            return str_replace($prefix, '', $key);
-        }, array_keys($filtered)), array_values($filtered));
+        return array_combine(array_map(fn(string $key) => (string)str_replace($prefix, '', $key), array_keys($filtered)), array_values($filtered));
```
Comment (a-laurowski) [inline src/Controller/Adminhtml/TestConnection/TestFtpConnection.php]:
```suggestion
        return array_combine(array_map(fn(string $key) => (string) str_replace($prefix, '', $key), array_keys($filtered)), array_values($filtered));
```

File: src/ViewModel/CategoryPath.php [hunk 2]
```diff
@@ -81,9 +81,7 @@ private function ngPath(?Category $category): array
 
     private function createPathFromCategory(?Category $category): array
     {
-        return array_map(function (Category $item): string {
-            return (string) $item->getName();
-        }, $category ? $category->getParentCategories() : []
+        return array_map(fn(Category $item) => (string) $item->getName(), $category ? $category->getParentCategories() : []
         );
```
Comment (a-laurowski) [inline src/ViewModel/CategoryPath.php]: this bracket could be moved to line 84, no ?

File: src/ViewModel/CategoryPath.php [hunk 3]
```diff
@@ -95,10 +93,7 @@ private function createPathFromCategory(?Category $category): array
     private function getParentCategories(?Category $category): array
     {
         $categories = $category ? $category->getParentCategories() : [];
-        usort($categories, function (Category $a, Category $b): int {
-            return $a->getLevel() - $b->getLevel();
-        }
-        );
+        usort($categories, fn(Category $a, Category $b) => $a->getLevel() - $b->getLevel());
```
Comment (a-laurowski) [inline src/ViewModel/CategoryPath.php]: missing returned type

**Review by a-laurowski — CHANGES_REQUESTED** [review]
sometimes we have
`fn ($args)`
and sometimes
`fn($args)`
please unify this.

What's more the arrow functions in many cases (if not all) don't contain returned type

File: src/ViewModel/CategoryPath.php (hunk 2 above)
Comment (paulfcdd) [reply src/ViewModel/CategoryPath.php]: done

**Comment by a-laurowski** [comment]
> sometimes we have `fn ($args)` and sometimes `fn($args)` please unify this.
> 
> What's more the arrow functions in many cases (if not all) don't contain returned type

I was looking for some PSR telling how to format the arrow function and i think we follow the closures rules
 
https://www.php-fig.org/psr/psr-12/ (7. Closures)
image
hence there should be space between `fn` and parenthesis

**Review by a-laurowski — CHANGES_REQUESTED** [review]
> sometimes we have `fn ($args)` and sometimes `fn($args)` please unify this.
> 
> What's more the arrow functions in many cases (if not all) don't contain returned type

I was looking for some PSR telling how to format the arrow function and i think we follow the closures rules
 
https://www.php-fig.org/psr/psr-12/ (7. Closures)
image
hence there should be space between `fn` and parenthesis

**Review by a-laurowski — APPROVED** [review]

**Pull request merged**